### PR TITLE
Convert doc builds to asciidoctor

### PR DESCRIPTION
**Pull request by karenzone**
Adds --asciidoctor flag to doc build script

### CODE REVIEW

**Comment by karenzone** [comment]
FYI: @jsvd @yaauie @nik9000  I think these changes are needed to use --asciidoctor for our docs automation. 

I have not tested, but I will.

File: build_docs.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ bundle exec ruby plugindocs.rb --output-path ../logstash-docs ../logstash/plugin
 
 cd ../logstash
 
-perl ../docs/build_docs --respect_edit_url_overrides --doc docs/index.asciidoc --resource=../logstash-docs/docs/ --chunk 1
+perl ../docs/build_docs --asciidoctor --respect_edit_url_overrides --doc docs/index.asciidoc --resource=../logstash-docs/docs/ --chunk 1
```
Comment (nik9000) [inline build_docs.sh]: Remove the `perl` bit here, I think. You just want to run `build_docs`. Perl doesn't need to get involved. `build_docs` happens to be python but it has a proper shebang so we have full control.

File: versioned_plugins.rb [hunk 2]
```diff
@@ -204,7 +204,7 @@ def test_docs
     puts "Cloning Docs repository"
     `git clone --depth 1 https://github.com/elastic/docs #{docs_path}`
     puts "Running docs build.."
-    `perl #{docs_path}/build_docs --doc #{logstash_docs_path}/docs/versioned-plugins/index.asciidoc --chunk 1`
+    `perl #{docs_path}/build_docs --asciidoctor --respect_edit_url_overrides --doc #{logstash_docs_path}/docs/versioned-plugins/index.asciidoc --chunk 1`
```
Comment (nik9000) [inline versioned_plugins.rb]: Same here.

**Comment by karenzone** [comment]
@nik9000 I pushed changes to address your comments.
I have tested changes with the manual plugin docs generation, and it works.

**Pull request merged**